[bitcode][dim] Assertion: should not be reached at calls.c:459

## Steps to Reproduce

1. `./autogen.sh --with-runtime-preset=bitcode --disable-boehm --enable-llvm`
2. (I had to put `<git_root>/llvm/usr/bin` on my PATH and also create a `clang++-6.0` symlink to `/usr/bin/clang` on OSX)
3. `make`
4. `make -C mono/tests test`


### Current Behavior

I get a crash when Mono is AOTing **dim-constrained3.exe** 

Stacktrace below

### Expected Behavior

Test compiles and runs.

## On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**:

I'm on a branch starting from 3ff64f1efb1

## Stacktrace

```
MONO_PATH=/Users/alklig/work/mono-pr-fixup/mono/mcs/class/lib/testing_aot_full ../../runtime/mono-wrapper --runtime=mobile --aot=llvmonly,clangxx=clang++-6.0,nimt-trampolines=2000,ntrampolines=10000,nrgctx-fetch-trampolines=256,ngsharedvt-trampolines=4400,nftnptr-arg-trampolines=4000 dim-constrained3.exe
Mono Ahead of Time compiler - compiling assembly /Users/alklig/work/mono-pr-fixup/mono/mono/tests/dim-constrained3.exe
AOTID 37551624-7A65-953C-D75B-20AAD7E20910
* Assertion: should not be reached at calls.c:459


=================================================================
	Native Crash Reporting
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x106559868 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mono_dump_native_crash_info
	0x1064d69a5 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mono_handle_native_crash
	0x106558dd1 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : sigabrt_signal_handler
	0x7fff7daaab5d - /usr/lib/system/libsystem_platform.dylib : _sigtramp
	0x107801900 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : _ZL10colorcodes
	0x7fff7d9646a6 - /usr/lib/system/libsystem_c.dylib : abort
	0x10673dfb7 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : monoeg_assert_abort
	0x10671e897 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mono_log_write_logfile
	0x10673e33e - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : monoeg_g_logv_nofree
	0x10673e4bf - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : monoeg_assertion_message
	0x10650180b - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mini_emit_method_call_full
	0x106460f68 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mono_method_to_ir
	0x10642c326 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mini_method_compile
	0x1064b1c84 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : compile_method
	0x1064a29c2 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mono_compile_assembly
	0x10649680f - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : mono_main
	0x106427088 - /Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono : main
	0x7fff7d8bf3d5 - /usr/lib/system/libdyld.dylib : start

=================================================================
	Telemetry Dumper:
=================================================================
No threads attached to runtime.

=================================================================
	External Debugger Dump:
=================================================================
(lldb) command source -s 0 '/tmp/mono-gdb-commands.20543'
Executing commands in '/tmp/mono-gdb-commands.20543'.
(lldb) process attach --pid 20543
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/copy.py", line 52, in <module>
    import weakref
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/weakref.py", line 14, in <module>
    from _weakref import (
ImportError: cannot import name _remove_dead_weakref
Process 20543 stopped
* thread #1, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x00007fff7d9f9356 libsystem_kernel.dylib`__wait4 + 10
libsystem_kernel.dylib`__wait4:
->  0x7fff7d9f9356 <+10>: jae    0x7fff7d9f9360            ; <+20>
    0x7fff7d9f9358 <+12>: movq   %rax, %rdi
    0x7fff7d9f935b <+15>: jmp    0x7fff7d9f4421            ; cerror
    0x7fff7d9f9360 <+20>: retq   
Target 0: (mono) stopped.

Executable module set to "/Users/alklig/work/mono-pr-fixup/mono/mono/mini/mono".
Architecture set to: x86_64h-apple-macosx-.
(lldb) thread list
Process 20543 stopped
* thread #1: tid = 0x2710776, 0x00007fff7d9f9356 libsystem_kernel.dylib`__wait4 + 10, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  thread #2: tid = 0x2710777, 0x00007fff7d9f786a libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'SGen worker'
(lldb) thread backtrace all
* thread #1, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fff7d9f9356 libsystem_kernel.dylib`__wait4 + 10
    frame #1: 0x0000000106559ab6 mono`mono_dump_native_crash_info at mini-posix.c:1079:4 [opt]
    frame #2: 0x0000000106559802 mono`mono_dump_native_crash_info(signal="SIGABRT", mctx=<unavailable>, info=<unavailable>) at mini-posix.c:1125 [opt]
    frame #3: 0x00000001064d69a5 mono`mono_handle_native_crash(signal="SIGABRT", mctx=0x00007ffee97d73a8, info=0x00007ffee97d7988) at mini-exceptions.c:3336:2 [opt]
    frame #4: 0x0000000106558dd1 mono`sigabrt_signal_handler(_dummy=6, _info=0x00007ffee97d7988, context=0x00007ffee97d79f0) at mini-posix.c:234:3 [opt]
    frame #5: 0x00007fff7daaab5d libsystem_platform.dylib`_sigtramp + 29
    frame #6: 0x00007fff7d9fa2c7 libsystem_kernel.dylib`__pthread_kill + 11
    frame #7: 0x00007fff7dab5bf1 libsystem_pthread.dylib`pthread_kill + 284
    frame #8: 0x00007fff7d9646a6 libsystem_c.dylib`abort + 127
    frame #9: 0x000000010673dfb7 mono`monoeg_assert_abort at goutput.c:57:3 [opt]
    frame #10: 0x000000010671e897 mono`mono_log_write_logfile(log_domain=<unavailable>, level=<unavailable>, hdr=<unavailable>, message="* Assertion: should not be reached at calls.c:459\n") at mono-log-common.c:136:3 [opt]
    frame #11: 0x000000010673e33e mono`monoeg_g_logv_nofree [inlined] monoeg_g_logstr(log_domain=0x0000000000000000, log_level=G_LOG_LEVEL_ERROR, msg=<unavailable>) at goutput.c:134:2 [opt]
    frame #12: 0x000000010673e315 mono`monoeg_g_logv_nofree(log_domain=<unavailable>, log_level=<unavailable>, format="* Assertion: should not be reached at %s:%d\n", args=<unavailable>) at goutput.c:149 [opt]
    frame #13: 0x000000010673e4bf mono`monoeg_assertion_message(format=<unavailable>) at goutput.c:184:22 [opt]
    frame #14: 0x000000010650180b mono`mini_emit_method_call_full(cfg=<unavailable>, method=0x00007fabffc3d6d0, sig=0x00007fac00038f78, tailcall=0, args=0x00007fac00038f78, this_ins=<unavailable>, imt_arg=0x0000000000000000, rgctx_arg=0x00007fac00039578) at calls.c:459:3 [opt]
    frame #15: 0x0000000106460f68 mono`mono_method_to_ir(cfg=<unavailable>, method=0x00007fabffc32c68, start_bblock=0x00007fac00038a18, end_bblock=0x00007fac00038b68, return_var=0x0000000000000000, inline_args=0x00007fac00038060, inline_offset=0, is_virtual_call=0) at method-to-ir.c:7760:11 [opt]
    frame #16: 0x000000010642c326 mono`mini_method_compile(method=<unavailable>, opts=391211519, domain=0x00007fabffc16fd0, flags=110, parts=0, aot_method_index=0) at mini.c:3437:2 [opt]
    frame #17: 0x00000001064b1c84 mono`compile_method(acfg=0x00007fac00028c00, method=0x00007fabffc32c68) at aot-compiler.c:8477:8 [opt]
    frame #18: 0x00000001064a29c2 mono`mono_compile_assembly [inlined] compile_methods(acfg=<unavailable>) at aot-compiler.c:11854:3 [opt]
    frame #19: 0x00000001064a275e mono`mono_compile_assembly(ass=0x00007fabffc33420, opts=<unavailable>, aot_options=<unavailable>, global_aot_state=0x00007ffee97d9260) at aot-compiler.c:13579 [opt]
    frame #20: 0x000000010649680f mono`mono_main at driver.c:1377:10 [opt]
    frame #21: 0x0000000106496779 mono`mono_main(argc=<unavailable>, argv=<unavailable>) at driver.c:2623 [opt]
    frame #22: 0x0000000106427088 mono`main [inlined] mono_main_with_options(argc=<unavailable>, argv=<unavailable>) at main.c:50:9 [opt]
    frame #23: 0x0000000106427074 mono`main(argc=6, argv=<unavailable>) at main.c:410 [opt]
    frame #24: 0x00007fff7d8bf3d5 libdyld.dylib`start + 1
    frame #25: 0x00007fff7d8bf3d5 libdyld.dylib`start + 1
  thread #2, name = 'SGen worker'
    frame #0: 0x00007fff7d9f786a libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff7dab656e libsystem_pthread.dylib`_pthread_cond_wait + 722
    frame #2: 0x0000000106715081 mono`thread_func [inlined] mono_os_cond_wait(cond=<unavailable>, mutex=<unavailable>) at mono-os-mutex.h:177:8 [opt]
    frame #3: 0x000000010671506e mono`thread_func at sgen-thread-pool.c:165 [opt]
    frame #4: 0x0000000106714fa3 mono`thread_func(data=0x0000000000000000) at sgen-thread-pool.c:196 [opt]
    frame #5: 0x00007fff7dab32eb libsystem_pthread.dylib`_pthread_body + 126
    frame #6: 0x00007fff7dab6249 libsystem_pthread.dylib`_pthread_start + 66
    frame #7: 0x00007fff7dab240d libsystem_pthread.dylib`thread_start + 13
(lldb) detach

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x7fff7d9fa2c6):0x7fff7d9fa2b6  ff ff c3 90 90 90 b8 48 01 00 02 49 89 ca 0f 05  .......H...I....
0x7fff7d9fa2c6  73 08 48 89 c7 e9 87 a1 ff ff c3 55 48 89 e5 41  s.H........UH..A
0x7fff7d9fa2d6  57 41 56 41 55 41 (lldb) quit
54 53 48 81 ec 88 00 00 00 4d  WAVAUATSH.Process 20543 detached
.....M
0x7fff7d9fa2e6  89 cf 49 89 f4 48 8d 05 76 a2 83 36 48 8b 00 48  ..I..H..v..6H..H
make[1]: *** [dim-constrained3.exe.dylib] Abort trap: 6
make: *** [test-recursive] Error 1
```

--- a/mono/mini/calls.c
+++ b/mono/mini/calls.c
@@ -436,9 +436,6 @@ mini_emit_method_call_full (MonoCompile *cfg, MonoMethod *method, MonoMethodSign
 	if (!sig)
 		sig = mono_method_signature_internal (method);
 
-	if (cfg->llvm_only && mono_class_is_interface (method->klass))
-		g_assert_not_reached ();
-
 	if (rgctx_arg) {
 		rgctx_reg = mono_alloc_preg (cfg);
 		MONO_EMIT_NEW_UNALU (cfg, OP_MOVE, rgctx_reg, rgctx_arg->dreg);

